File: python/Update.py
          
def update_student():
        import os
        id=input("Enter the ID of the record\n")

        file= open(path,"r")  
        temp=open("tempfile.txt",'w')
        flag=False
        for line in file:
            st=line.split("\t")

            if st[0]==id:
                flag=True
                name=input("Please enter the new name \n")
                age =input("please enter the new age \n")
                line =id+'\t'+name+'\t'+age+'\n'
        temp.write(line)
        
        file.close()
        temp.close()
        os.remove(path)
        os.rename("tempfile.txt",path)
        if not flag:
          print("No such element")     
        else :
            print("UPDATED")  
                
update_student()

File: CPP/Update.cpp
 //         اللهم لا سهل الا ما جعلته سهلا و انت تجعل الصعب ان شئت سهل        
 #include <bits/stdc++.h> 
  #define ll                            long long 
  #define nl                            "\n"

 using namespace std;

class student{
    public:
    int id,age;
    char name[10];
};
string path_student="D:\\data.txt";


void update_student(){
fstream f;
student s;
f.open(path_student,ios::in|ios::out);

 char name[10];
cout<<"Please enter name to update"<<nl;
cin >> name;
if ( f.is_open()){
while ( f.read((char *)&s,sizeof(s))){

if ( strcmp(name,s.name)==0){
    cout<<"Please enter the new age "<<nl;
    cin >> s.age;
   f.write((char *)&s,sizeof(s));
    f.close();
    return ;

}

}

}else {
    cout<<"This file is not found "<<nl;
}



}



int main(){
update_student();
  return 0;
}
